src/protocol.rs

```diff
@@ -407,6 +416,132 @@ mod tests {
     use std::collections::HashMap;
     use tracing::debug;
 
+    //#[cfg_attr(feature = "flame_it", flame)]
+    #[test]
+    fn process_single_message_wrong_session_id() -> Result<()> {
+        let mut rng = init_testing();
+        let QUORUM_SIZE = 3;
+        // Set GLOBAL config for participants
+        let configs = ParticipantConfig::random_quorum(QUORUM_SIZE, &mut rng);
+
+        // Set up auxinfo participants
+        let auxinfo_sid = Identifier::random(&mut rng);
+        let mut auxinfo_quorum = configs
+            .iter()
+            .map(|config| Participant::<AuxInfoParticipant>::from_config(config, auxinfo_sid, ()))
+            .collect::<Vec<_>>();
+
+        let message_type = MessageType::Auxinfo(crate::messages::AuxinfoMessageType::Public);
+        let mut rng1 = init_testing();
+        let identifier = Identifier::random(&mut rng1);
+        let from = auxinfo_quorum[0].id;
+        let to = auxinfo_quorum[0].id;
+        let unverified_bytes = vec![0x01, 0x02, 0x03];
+        let message = Message::new(message_type, identifier, from, to, &unverified_bytes);
+        let participant = auxinfo_quorum.iter_mut().choose(&mut rng).unwrap();
+
+        if let Err(e) = participant.process_single_message(&message, &mut rng) {
+            assert_eq!(
+                e,
+                InternalError::CallingApplicationMistake(CallerError::WrongSessionId)
+            );
+        };
+        Ok(())
+    }
+
+    #[test]
+    fn process_single_message_wrong_participant_to_field() -> Result<()> {
+        let mut rng = init_testing();
+        let QUORUM_SIZE = 3;
+        // Set GLOBAL config for participants
+        let configs = ParticipantConfig::random_quorum(QUORUM_SIZE, &mut rng);
+
+        // Set up auxinfo participants
+        let auxinfo_sid = Identifier::random(&mut rng);
+        let mut auxinfo_quorum = configs
+            .iter()
+            .map(|config| Participant::<AuxInfoParticipant>::from_config(config, auxinfo_sid, ()))
+            .collect::<Vec<_>>();
+
+        let message_type = MessageType::Auxinfo(crate::messages::AuxinfoMessageType::Public);
+        let from = auxinfo_quorum[0].id;
+        let to = auxinfo_quorum[0].id;
+        let unverified_bytes = vec![0x01, 0x02, 0x03];
+        let message = Message::new(message_type, auxinfo_sid, from, to, &unverified_bytes);
+        let participant = auxinfo_quorum.iter_mut().choose(&mut rng).unwrap();
+        if let Err(e) = participant.process_single_message(&message, &mut rng) {
```

Comment:
todo: This test shouldn't pass consistently. We choose a random participant from the quorum here and set the `to` field to also be a valid participant ID from the quorum, so I'd expect this to pass about a third of the time. I would recommend intentionally choosing a (participant, ID) pair that don't match, so that this will always fail as expected.

I think this passes because of the comment above -- if `process_single_message` returns `Ok(...)`, we skip the assert and the test passes.